File: src/main/java/dk/cngroup/fetch/entity/Product.java
package dk.cngroup.fetch.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.validator.constraints.Length;
import org.springframework.data.rest.core.annotation.Description;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotNull;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Product extends AuditableEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @NotNull
    @Length(min = 3, max = 30)
    @Description("Unique name of the item")
    String code;

    public Product(String code) {
        this.code = code;
    }
}

File: src/main/java/dk/cngroup/fetch/entity/Product.kt
package dk.cngroup.fetch.entity

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.rest.core.annotation.Description
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.validation.constraints.NotNull

@Entity
class Product(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Description("Unique name of the item")
    @field:NotNull
    val code: String
) : AuditableEntity()

interface ProductRepository : JpaRepository<Product?, Long?>